samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -84,6 +84,7 @@ private String getHostName() {
 
   @Override
   public void tryBecomeLeader(LeaderElectorListener leaderElectorListener) {
+    zkUtils.makeSurePersistentPathsExists(new String[]{keyBuilder.getProcessorsPath()});
```

Comment:
Agree with @xinyuiscool when we make the Zk coordination as a generic service, leader election and other job-coordinator specific functions should be separated. Let's open a JIRA to cleanup in another PR.